Add Model.record_exists helper method

dan-kez opened the pull request:
I recently added this helper method in my own project and thought it may be useful upstream. In short, this adds a short form utility to check if a record exists by looking up the hash_key and optional range_key.

I'm less familiar with how the tests are set up here and attempted to add both an integration and unit test. If there is something I could do better here please let me know!

ikonst commented on pynamodb/models.py:
```diff
@@ -675,6 +675,26 @@ def scan(cls,
         )
 
+    @classmethod
+    def record_exists(cls, hash_key, range_key=None):
```
If we add this method, we should probably update models.pyi with its type information.

dan-kez replied:
Sounds good!

ikonst commented on pynamodb/models.py:
```diff
@@ -675,6 +675,26 @@ def scan(cls,
+        :param range_key: If set, include in lookup
+        """
+        try:
+            if range_key is None:
+                cls.query(hash_key).next()
+            else:
+                cls.query(hash_key, cls._range_key_attribute() == range_key).next()
```
```suggestion
            if range_key is None:
                next(cls.query(hash_key, page_size=1))
            else:
                next(cls.query(hash_key, cls._range_key_attribute() == range_key, page_size=1))
```

1. I recall restricting `page_size` leads to spending fewer RCUs. @garrettheel?
2. Total nit, but maybe using `next` built-in is more idiomatic?

dan-kez replied:
Sounds good to me, I applied your suggestion.

dan-kez commented:
@ikonst I just addressed your comments and added an additional kwarg, `range_key` for, what I think is the most common use case of direct equality comparisons. 

In the doc string I indicated that `range_key` would have priority over `range_key_condition` if they are both present. If you would rather me throw an error instead of allowing this let me know. I'm happy to throw a `ValueError` or something

ikonst commented on pynamodb/models.py:
```diff
@@ -675,6 +675,33 @@ def scan(cls,
         )
 
+    @classmethod
+    def record_exists(cls, hash_key, range_key_condition=None, range_key=None):
```
Maybe a way to avoid the "what if both arguments are passed" problem would be to have just one argument and branch on `isinstance(range_key, Condition)`?

dan-kez replied:
So this does match your current argument structure and naming where the second key is the condition expression (see `query`)

I am happy to do that branching mechanism but was concerned that it would be a divergent API compared to query and others (e.g. `MyModel.query('hash_key', 'range_key'` will not work today).

If you'd like me still reorder, rename, or remove something let me know!

ikonst replied:
Yeah, it's `query(hash_key, range_key_condition=None)`. I guess we don't have a `query(hash_key, range_key)` shortcut form, but we want to have a shortcut form here?

dan-kez replied:
I'm happy to remove it if you'd rather not have it. In all my use cases I always want to check if a single record exists by looking up both the hash_key and range_key directly. The short form is nice since since i don't need to write out the whole range condition.

All that said, I'm happy to just drop the short form
